### PR TITLE
tests: Port AVM1 `Math` tests to our test framework

### PR DESCRIPTION
**Pull request by moulins**
This was one of the last AVM1 tests implemented directly in Rust.

Changes:
- Add `approximations.bare_numbers` test config key, to disable approximations on number-only lines
- Remove SWFv5 tests, as MTASC cannot compile them (and FP doesn't appear to have any
  SWFv5-specific behavior here)
- Test more edge-cases, exposing some incorrect Ruffle output

### CODE REVIEW

File: tests/framework/src/options/approximations.rs [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+use anyhow::bail;
 use approx::relative_eq;
 use regex::Regex;
 use serde::Deserialize;
 
 #[derive(Clone, Deserialize, Default)]
 #[serde(default, deny_unknown_fields)]
 pub struct Approximations {
+    #[serde(default = "default_for_bare_numbers")]
+    pub bare_numbers: bool,
```
Comment (kjarosh) [inline tests/framework/src/options/approximations.rs:10]: Do we want to make it true by default? TBH I wasn't aware that the test framework will compare bare numbers in addition to patterns.

Comment (moulins) [reply tests/framework/src/options/approximations.rs:10]: Switching it to false-by-default would mean changing a lot of existing tests; it's probably best left to a separate PR if we decide this is something we want to do.

File: tests/tests/swfs/avm1/math_swf8/Test.as [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    trace(invalid());
+    trace("// ASnative(200, 50)({ v: 3 }, { v: 2 }, { v: 1 })");
+    trace(invalid(noisy(3), noisy(2), noisy(1)));
+
+    fscommand("quit");
+  }
+
+  static var NOISY_VALUE_OF = function() {
+    trace("valueOf called: " + this.v);
+    return this.v;
```
Comment (kjarosh) [inline tests/tests/swfs/avm1/math_swf8/Test.as:106]: Can you add test cases where this throws?

Comment (moulins) [reply tests/tests/swfs/avm1/math_swf8/Test.as:106]: Throwing coercions are already tested in `avm1/displacementmapfilter_mappoint_throw_error`, so I don't think this is necessary (in particular, `Math` has no internal state that could be partially modified before one of the coercions throws).

Comment (kjarosh) [reply tests/tests/swfs/avm1/math_swf8/Test.as:106]: I don't see the connection, you have setters and different impls there. I'm not talking about testing that coercions throw exceptions (we know they do), but how the given methods handle them.

How do you know how exceptions should be handled in Math? Will the second coercion run before the first is rethrown? If both throw, which one is rethrown? Does it rethrow at all?

Comment (moulins) [reply tests/tests/swfs/avm1/math_swf8/Test.as:106]: It would be weird if it did anything else than immediately rethrow the first exception, but I guess I can add a test for that
